crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -0,0 +1,536 @@
+use crate::{
+    service::{NamespaceIdentUuid, TableIdentUuid},
+    WarehouseIdent,
+};
+
+use crate::api::{iceberg::v1::DataAccess, CatalogConfig, ErrorModel, Result};
+use crate::service::storage;
+use crate::service::storage::path_utils::reduce_scheme_string;
+use crate::service::tabular_idents::TabularIdentUuid;
+use azure_storage::prelude::{BlobSasPermissions, BlobSignedResource};
+use azure_storage::shared_access_signature::service_sas::BlobSharedAccessSignature;
+use azure_storage::shared_access_signature::SasToken;
+use azure_storage::StorageCredentials;
+use futures::StreamExt;
+use iceberg::io::azdls;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::SystemTime;
+use url::Url;
+use uuid::Uuid;
+use veil::Redact;
+
+fn is_valid_bucket_name(bucket: &str) -> Result<()> {
+    // Bucket names must be between 3 (min) and 63 (max) characters long.
+    if bucket.len() < 3 || bucket.len() > 63 {
+        return Err(ErrorModel::bad_request(
+            "Bucket name must be between 3 and 63 characters long.",
+            "InvalidBucketName",
+            None,
+        )
+        .into());
+    }
+
+    // Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).
```

Comment:
Are we sure dots can be in there?

I am getting:
This name may only contain lowercase letters, numbers, and hyphens, and must begin with a letter or a number. Each hyphen must be preceded and followed by a non-hyphen character. The name must also be between 3 and 63 characters long.